test/AutoDiff/differentiable_attr_type_checking.swift

```diff
@@ -590,12 +590,12 @@ protocol DiffReq : Differentiable {
 
 // expected-error @+1 {{does not conform to protocol 'DiffReq'}}
 struct ConformingWithErrors : DiffReq {
-  // expected-note @+1 {{candidate is missing attribute '@differentiable(wrt: (self, x))'}}
+  // expected-note @+1 {{candidate is missing attribute '@differentiable'}}
```

Comment:
The PR description says it's maintaining the same printing behavior, but it seems not to be the case. We used `ParsedParameterIndices` because diagnostics should be WYSIWIG from the protocol declaration. Is there a way to still maintain that behavior?

Reply:
We discussed in person.

We agreed that the new printing behavior is desirable, because the printed `@differentiable` is shorter. However, it's necessary to account for cross-module differences in parameter type differentiability; it should be valid to directly use the printed `@differentiable` attribute.

Done in https://github.com/apple/swift/pull/23379/commits/c989899bdbe0c073500ff407a56bbe9d468702da. Cross-module test added.